### ISSUE TITLE
Why doesn't the shipping method appear on the checkout page when running using npm run dev?

### ISSUE THREAD

**Issue by ekakrisna** [wix/app-template-custom-shipping-rates#20]
Hi, thanks for sharing this repository. I'm encountering an issue when running it on localhost with npm run dev as the shipping method does not appear on the checkout page. I've followed the provided instructions, but could you assist me in running it on my localhost?
![image](https://github.com/wix/app-template-custom-shipping-rates/assets/37111833/4d84ff32-06c4-4665-bc1f-a56cca4384c3)

**Comment by carmelc** [comment]
Hi, in order to see the custom shipping rate the commerce server has to invoke the SPI, and it cannot call a local machine (localhost:3000) address, so the code has to be deployed on a server and the shipping rate has to be switched on for the geo in order to see it in the cart page and checkout,

**Comment by ekakrisna** [comment]
Hi, do you know how to display shipping rates locally?

**Comment by ekakrisna** [comment]
Yes, I have already deployed to production, but it still doesn't work. Could you perhaps provide instructions on how to set this up?
![Screenshot 2024-05-16 092935](https://github.com/wix/app-template-custom-shipping-rates/assets/37111833/f89a7db3-6b76-44c3-badd-6d7ec9d9d428)
![image](https://github.com/wix/app-template-custom-shipping-rates/assets/37111833/f10e62ab-67b3-41b4-8bb0-c85bf4ac5e28)

**Comment by carmelc** [comment]
Hi,

Could you please confirm if all the provided URLs are valid and accessible? Specifically, have you deployed them on a provider that supports Next.js 14, and are you able to invoke them successfully?

When you mention that they do not work, are you referring to the shipping rate SPI, the dashboard page, or both? If you are unable to see the new shipping rate, have you configured it as one of the shipping rates in your store? You can refer to this guide for assistance: [Wix Stores: A Guide to Setting Up Shipping, Delivery, and Pickup](https://support.wix.com/en/article/wix-stores-a-guide-to-setting-up-shipping-delivery-and-pickup).

Please note that the application you created does not add a new shipping rate to all new orders but rather provides a shipping rate option that you can associate with a region.

Thank you.

**Comment by ekakrisna** [comment]
For now, the issue here is that the SPI shipping rate does not work when I create a new app. I have configured it according to the reference you provided, but it still doesn't work.
![image](https://github.com/wix/app-template-custom-shipping-rates/assets/37111833/d9ba2616-f794-4f81-aa94-80ba17814494)
![image](https://github.com/wix/app-template-custom-shipping-rates/assets/37111833/ce3d9d85-272e-4724-aa39-99efa9b07403)

**Comment by carmelc** [comment]
Make sure that the currency you defined in your site is the same as the one returned by the SPI

**Comment by txemaquin** [comment]
@ekakrisna After days of trying to make this template work, I found out that deploymentUri should be {BASEURL}/api/shipping-rates/ in the service JSON file, hope this can help you.

**Comment by bizmate** [comment]
for a static url on your machines use Ngrok

**Comment by txemaquin** [comment]
hi,

sorry for not telling you before, all is working properly now. We are
finishing the rest.

Thanks

**Comment by carmelc** [comment]
Great to hear :)
